spec/libraries/course/conditional/conditional_dependency_graph_spec.rb

```diff
@@ -0,0 +1,229 @@
+require 'rails_helper'
+
+RSpec.describe Course::Conditional::ConditionalDependencyGraph do
+  class DummyConditionalCondition < ActiveRecord::Base
+    acts_as_conditional
+    acts_as_condition
+    attr_accessor :conditions
+    attr_accessor :id
+
+    def specific_conditions
+      @conditions
+    end
+
+    def dependent_object
+      self
+    end
+
+    def inspect
+      "<#{@id}>"
+    end
+
+    def self.columns
+      []
+    end
+
+    def self.build(conditions, id)
+      dummy = new
+      dummy.conditions = conditions
+      dummy.id = id
+      dummy
+    end
+  end
+
+  def create_simple_graph
+    #
+    # A ---------
+    #    |   |   |
+    #    v   v   |
+    #    C-->D   |
+    #        |   |
+    #        ->E<-
+    # B -------^
+    #
+    @a = DummyConditionalCondition.build([], 'A')
```

Comment:
we don't use member variables in specs. perhaps you can try to encapsulate the entire graph in an object and pass that around instead?

you can also just return a hash of the nodes?